nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/json/AbstractJsonRowRecordReader.java

```diff
@@ -95,6 +100,41 @@ public AbstractJsonRowRecordReader(final InputStream in, final ComponentLog logg
         }
     }
 
+    protected AbstractJsonRowRecordReader(final InputStream in, final ComponentLog logger, final String dateFormat, final String timeFormat, final String timestampFormat,
+                                          final String skipToNestedJsonField) throws IOException, MalformedRecordException {
+
+        this(logger, dateFormat, timeFormat, timestampFormat);
+
+        try {
+            jsonParser = jsonFactory.createParser(in);
+            jsonParser.setCodec(codec);
+
+            if (skipToNestedJsonField != null) {
+                while (!jsonParser.nextFieldName(new SerializedString(skipToNestedJsonField))) {
+                    // go to nested field if specified
+                    if (!jsonParser.hasCurrentToken()) {
+                        throw new IOException("The defined skipTo json field is not found when processing json as NiFi record.");
+                    }
```

Comment:
Is this check necessary inside the loop? It seems like the check should be moved outside the loop, or it could be removed completely since `jsonParser.nextToken()` later on checks that the token is not `null`.

Reply:
It was necessary to break the loop when the skipToFieldName was not found. I changed it to:

`while (!jsonParser.nextFieldName(serializedSkipToField) && jsonParser.hasCurrentToken()) {
 }`

And the implementation will follow the original behaviour aka return null when the field is not found.